consumers/google.go

```diff
@@ -172,31 +177,40 @@ func (d *googleDNSConsumer) applyChange(change *dns.Change) error {
 	return nil
 }
 
-func (d *googleDNSConsumer) currentRecords() (map[string]*dns.ResourceRecordSet, error) {
+func (d *googleDNSConsumer) currentRecords() (map[string]*ownedRecord, error) {
 	resp, err := d.client.ResourceRecordSets.List(params.project, params.zone).Do()
 	if err != nil {
 		return nil, fmt.Errorf("Error getting DNS records from %s/%s: %v", params.project, params.zone, err)
 	}
 
-	records := make(map[string]*dns.ResourceRecordSet)
+	records := make(map[string]*ownedRecord)
 
 	for _, r := range resp.Rrsets {
-		if r.Type == "TXT" {
-			records[r.Name] = r
-		} else {
-			if _, exists := records[r.Name]; !exists {
-				records[r.Name] = nil
+		if r.Type == "A" || r.Type == "TXT" {
+			record := records[r.Name]
+
+			if record == nil {
+				record = &ownedRecord{}
 			}
+
+			switch r.Type {
+			case "A":
```

Comment:
if i am not mistaken, this may result in existing records being overwritten ? (because they are not added to the map)